ros2_foxglove_bridge/src/message_definition_cache.cpp

```diff
@@ -211,59 +224,89 @@ const MessageSpec& MessageDefinitionCache::load_message_spec(
-      }
 
-      // Find the the subtype that was originally requested and return it.
-      const auto it = msg_specs_by_definition_identifier_.find(definition_identifier);
-      if (it == msg_specs_by_definition_identifier_.end()) {
-        throw DefinitionNotFoundError(definition_identifier.package_resource_name);
-      }
-      return it->second;
-    } else {
-      RCUTILS_LOG_ERROR_NAMED("foxglove_bridge",
-                              "Action IDL definitions are currently not supported");
+  if (subfolder == "action") {
+    const auto splitDefinitions = foxglove_bridge::splitMessageDefinitions(file);
+    if (splitDefinitions.size() != 3) {
+      throw std::invalid_argument("Invalid action definition: " + filename);
+    }
+
+    const auto& goalDef = splitDefinitions[0];
+    const auto& resultDef = splitDefinitions[1];
+    const auto& feedbackDef = splitDefinitions[2];
+
+    // Define type definitions for each action subtype.
+    // These type definitions may include additional fields such as the goal_id.
+    // See also https://design.ros2.org/articles/actions.html
+    const std::map<std::string, std::string> action_type_definitions = {
+      {ACTION_FEEDBACK_MESSAGE_SUFFIX, "unique_identifier_msgs/UUID goal_id\n" + feedbackDef},
+      {std::string(ACTION_RESULT_SERVICE_SUFFIX) + SERVICE_REQUEST_MESSAGE_SUFFIX,
+       "unique_identifier_msgs/UUID goal_id\n"},
+      {std::string(ACTION_RESULT_SERVICE_SUFFIX) + SERVICE_RESPONSE_MESSAGE_SUFFIX,
+       "int8 status\n" + resultDef},
+      {std::string(ACTION_GOAL_SERVICE_SUFFIX) + SERVICE_REQUEST_MESSAGE_SUFFIX,
+       "unique_identifier_msgs/UUID goal_id\n" + goalDef},
+      {std::string(ACTION_GOAL_SERVICE_SUFFIX) + SERVICE_RESPONSE_MESSAGE_SUFFIX,
+       "bool accepted\nbuiltin_interfaces/msg/Time stamp"}};
+
+    // Create a MessageSpec instance for every action subtype and add it to the cache.
+    const std::string action_name = remove_action_subtype(type_name);
+    for (const auto& [action_suffix, definition] : action_type_definitions) {
+      DefinitionIdentifier definition_id;
+      definition_id.format = definition_identifier.format;
+      definition_id.package_resource_name = package + "/action/" + action_name + action_suffix;
+      msg_specs_by_definition_identifier_.emplace(
+        definition_id, MessageSpec(definition_id.format, definition, package));
+    }
+
+    // Find the the subtype that was originally requested and return it.
+    const auto it = msg_specs_by_definition_identifier_.find(definition_identifier);
+    if (it == msg_specs_by_definition_identifier_.end()) {
+      throw DefinitionNotFoundError(definition_identifier.package_resource_name);
+    }
+    return it->second;
+  } else if (subfolder == "srv") {
+    const auto splitDefinitions = foxglove_bridge::splitMessageDefinitions(file);
+    if (splitDefinitions.size() != 2) {
+      throw std::invalid_argument("Invalid service definition: " + filename);
```

Comment:
nit: would be helpful to include the expected & actual number of definitions for debugging